projects/ng-components/lib/toggle-selector/toggle-selector.component.html

```diff
@@ -0,0 +1,28 @@
+<ng-container *ngFor="let item of displayList; let idx = index; trackBy: trackByFn">
+    <div
+        class="ac toggle-selector"
+        [class.error]="error"
+        [class.checked]="item.selected"
+        [class.disabled]="disabled || item.optionItem.disabled"
+        [class.focus]="focusedItem === idx"
+        (click)="handleItemOnClick($event, idx)"
+    >
+        <input
+            [type]="inputType"
+            [attr.aria-label]="item.optionItem.label"
+            [id]="item.optionItem.value"
+            [name]="name"
+            [disabled]="disabled || item.optionItem.disabled"
+            (focus)="focus(idx)"
+            (blur)="blur()"
```

Comment:
Wouldn't this be easier to maintain? since it's just a simple operation we don't need to go back and forth between the template and the javascript part.
```suggestion
            (focus)="focusedItem = idx"
            (blur)="focusedItem = null"
```

Reply:
let's not write JS operation inside HTML files, it get's very confusing when the component is huge.

Reply:
this is very similar to:
```html
<div [ngClass]="{ isDisabled: disabled, theme: theme || 'primary' }"></div>
```
The `ng-class` directive processes JavaScript logic